cpp/src/replace/nulls.cu

```diff
@@ -367,7 +366,7 @@ std::unique_ptr<cudf::column> replace_nulls_scalar_kernel_forwarder::operator()<
 std::unique_ptr<cudf::column> replace_nulls_policy_impl(cudf::column_view const& input,
                                                         cudf::replace_policy const& replace_policy,
                                                         rmm::cuda_stream_view stream,
-                                                        rmm::mr::device_memory_resource* mr)
+                                                        rmm::mr::device_memory_resource*)
```

Comment:
this is a bug. we should be using the provided mr to allocate the return value.

Reply:
Yea, it should be propagated down to the `cudf::detail::gather` call. I will open a PR that fixes this issue

Reply:
https://github.com/rapidsai/cudf/pull/8500